include/dali/core/mm/detail/align.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_MM_DETAIL_ALIGN_H_
+#define DALI_CORE_MM_DETAIL_ALIGN_H_
+
+#include <cassert>
+#include <utility>
+#include "dali/core/util.h"
+
+namespace dali {
+namespace mm {
+namespace detail {
+
+DALI_HOST_DEV
+constexpr uintptr_t ptr2u(void *p) noexcept {
+  return static_cast<char*>(p) - static_cast<char*>(nullptr);
+}
+
+template <typename T>
+DALI_HOST_DEV
+constexpr T *u2ptr(uintptr_t i) noexcept {
+  return static_cast<T*>(static_cast<void *>(static_cast<char*>(nullptr) + i));
+}
```

Comment:
Apparently, `reinterpret_cast<T*>(i)` is not a constant expression - but this series of static casts is.

Reply:
Same as above.